server/src/main/java/com/cloud/network/lb/LoadBalancingRulesManagerImpl.java

```diff
@@ -1598,65 +1602,73 @@ public LoadBalancer createPublicLoadBalancerRule(String xId, String name, String
         // LoadBalancer result = _elbMgr.handleCreateLoadBalancerRule(lb,
         // lbOwner, lb.getNetworkId());
         LoadBalancer result = null;
-        if (result == null) {
-            IpAddress systemIp = null;
-            NetworkOffering off = _entityMgr.findById(NetworkOffering.class, network.getNetworkOfferingId());
-            if (off.isElasticLb() && ipVO == null && network.getVpcId() == null) {
-                systemIp = _ipAddrMgr.assignSystemIp(networkId, lbOwner, true, false);
-                if (systemIp != null) {
-                    ipVO = _ipAddressDao.findById(systemIp.getId());
+        IpAddress systemIp = null;
+        NetworkOffering off = _entityMgr.findById(NetworkOffering.class, network.getNetworkOfferingId());
+
+        if (srcPortStart == 53) {
+            List<NetworkOfferingServiceMapVO> offerinServices = _networkOfferingServiceDao.listByNetworkOfferingId(network.getNetworkOfferingId());
+            for (NetworkOfferingServiceMapVO serviceMapVo: offerinServices){
+                if (serviceMapVo.getService() == Service.Dns.getName()) {
+                    throw new InvalidParameterValueException("Error adding load balancer rule, cannot add port 53 with network service offering having DNS service.");
```

Comment:
Loads of false positives on the diff. This is actually the only change @Spaceman1984 , or am I missing something else?

Reply:
Yes @DaanHoogland , mostly reformatting, there was an unnecessary check which I removed:
```
 LoadBalancer result = null;
 if (result == null) {
```